modules/dynamic-operator/plugin/commons/interface.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package commons
+
+import (
+	"github.com/hashicorp/go-hclog"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// PluginService is the interface that we're exposing as a plugin.
+type PluginService interface {
+	SetLogger(logger hclog.Logger)
+	Empty(req PluginRequest) *PluginResponse
+	ForCreate(req PluginRequest) *PluginResponse
+	ForUpdate(req PluginRequest) *PluginResponse
+	Status(req PluginRequest) *PluginResponse
+	Type() *PluginResponse
```

Comment:
- Do we even need this one if we have Empty() / New() methods?
- Type() need to enabling watching to the service. In that case we do not have any name/namespace and other services